src/components/AddNewBook.js

```diff
@@ -1,14 +1,59 @@
-const AddNewBook = () => (
-    <form className="form-inline">
-        <label className="sr-only" htmlFor="inlineFormInputName2"></label>
-        <input type="text" className="form-control mb-2 mr-sm-2" id="inlineFormInputName2" placeholder="Book Title" />
+import { useDispatch, useSelector } from 'react-redux';
+import BookModel from '../models/bookmodel';
+import {
+  ADD_BOOK, bookAction, EDIT_NEWBOOK, EDIT_NEWBOOK_AURTHOR,
+} from '../redux/books/books';
 
-        <label className="sr-only" htmlFor="inlineFormInputGroupUsername2"></label>
-        <div className="input-group mb-2 mr-sm-2">
-            <input type="text" className="form-control" id="inlineFormInputGroupUsername2" placeholder="Author" />
-        </div>
+const AddNewBook = () => {
+  const dispatch = useDispatch();
+  const newBook = useSelector((state) => state.books.newBook);
+  return (
+        <form className="form-inline">
+            <label
+                className="sr-only"
+                htmlFor="inlineFormInputName2"
+            />
+            <input
+                type="text"
+                className="form-control mb-2 mr-sm-2"
+                id="inlineFormInputName2"
+                placeholder="Book Title"
+                value={newBook.bookName}
+                onChange={(e) => dispatch(bookAction({
+                  type: EDIT_NEWBOOK,
+                  bookName: e.target.value,
+                }))}
+            />
 
-        <button type="submit" className="btn btn-primary mb-2">Add Book</button>
-    </form>
-);
+            <label
+                className="sr-only"
+                htmlFor="inlineFormInputGroupUsername2"
+            />
+            <div className="input-group mb-2 mr-sm-2">
+                <input
+                    type="text"
+                    className="form-control"
+                    id="inlineFormInputGroupUsername2"
+                    placeholder="Author"
+                    value={newBook.author}
+                    onChange={(e) => dispatch(bookAction({
+                      type: EDIT_NEWBOOK_AURTHOR,
+                      author: e.target.value,
+                    }))}
+                />
+            </div>
+
+            <button
+                type="submit"
+                className="btn btn-primary mb-2"
+                onClick={(e) => {
+                  e.preventDefault();
+                  const tempBookModel = new BookModel(newBook.bookName, newBook.author, 'Testing');
+                  dispatch(bookAction(
+                    { type: ADD_BOOK, book: JSON.stringify(tempBookModel) },
+                  ));
```

Comment:
### You were expected to have:
- [ ] A local state with  title, author where you should have your inputs update the local state every time the inputs are updated (onChange)
- [ ] A `submitBookToStore` that takes the inputs from the state and generate a unique `id` and add them to an object, then dispatches the `addBook` action creator (name of the function can be different) with the object as its argument